uwblueprint/planet-read
# Fix insert_test_data.py

## Notion ticket link





## Implementation description
* Updated `insert_test_data.py` to only insert if entry does not already exist
* Added erase key to allow database to be cleared using the script



## Steps to test
1. Rebuild all containers
2. Run `docker exec -it planet-read_py-backend_1 /bin/bash -c "flask db upgrade"`
3. Check that the tables are created properly ` docker exec -it planet-read_db_1 mysql -u root -proot -e "USE planet-read; SHOW TABLES;"`
3. Run `docker exec -it planet-read_py-backend_1 /bin/bash -c "python -m tools.insert_test_data"`
4. Check that database is seeded: e.g. `docker exec -it planet-read_db_1 mysql -u root -proot -e "USE planet-read; SELECT * from comments;"`
5. Try seeding the database again: `docker exec -it planet-read_py-backend_1 /bin/bash -c "python -m tools.insert_test_data"`
6. Check that duplicate entries are not added into the tables: e.g. `docker exec -it planet-read_db_1 mysql -u root -proot -e "USE planet-read; SELECT * from users;"`(this should only show 7 users)
7. Try erasing the database:  `docker exec -it planet-read_py-backend_1 /bin/bash -c "python -m tools.insert_test_data erase"`
8. Check that the database is now emptied: e.g. `docker exec -it planet-read_db_1 mysql -u root -proot -e "USE planet-read; SELECT * from users;"`
9. Downgrade the database:  `docker exec -it planet-read_py-backend_1 /bin/bash -c "flask db downgrade base"`
10. Check that there is not current revision: `docker exec -it planet-read_py-backend_1 /bin/bash -c "flask db current --verbose"`


## What should reviewers focus on?
* Updates to `insert_test_data.py`


## Checklist
- [x] For backend changes, I have run the appropriate linters:  `docker exec -it <backend-container-id> /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
